BitwardenShared/Core/Platform/Services/KeyConnectorService.swift

```diff
@@ -39,27 +49,53 @@ class DefaultKeyConnectorService {
     /// Initialize a `DefaultKeyConnectorService`.
     ///
     /// - Parameters:
+    ///   - accountAPIService: The service used by the application to make account related API requests.
+    ///   - clientService: The service that handles common client functionality such as encryption
+    ///     and decryption.
     ///   - keyConnectorAPIService: The API service used to make key connector requests.
     ///   - stateService: The service used by the application to manage account state.
     ///
     init(
+        accountAPIService: AccountAPIService,
+        clientService: ClientService,
         keyConnectorAPIService: KeyConnectorAPIService,
         stateService: StateService
     ) {
+        self.accountAPIService = accountAPIService
+        self.clientService = clientService
         self.keyConnectorAPIService = keyConnectorAPIService
         self.stateService = stateService
     }
 }
 
 extension DefaultKeyConnectorService: KeyConnectorService {
-    func getMasterKeyFromKeyConnector() async throws -> String {
+    func convertNewUserToKeyConnector(keyConnectorUrl: URL, orgIdentifier: String) async throws {
+        let keyConnectorResponse = try await clientService.auth().makeKeyConnectorKeys()
+
+        try await keyConnectorAPIService.postMasterKeyToKeyConnector(
+            key: keyConnectorResponse.masterKey,
+            keyConnectorUrl: keyConnectorUrl
+        )
+
         let account = try await stateService.getActiveAccount()
-        let keyConnectorUrlString = account.profile.userDecryptionOptions?.keyConnectorOption?.keyConnectorUrl
-        guard let keyConnectorUrlString,
-              let keyConnectorUrl = URL(string: keyConnectorUrlString) else {
-            throw KeyConnectorServiceError.missingKeyConnectorUrl
-        }
+        try await accountAPIService.setKeyConnectorKey(
+            SetKeyConnectorKeyRequestModel(
+                kdfConfig: account.kdf,
```

Comment:
🤔 In MAUI the KDF config is set from the `IdentityTokenResponse` not from the current `Account`. Not sure if it makes a difference though as it should be the same but wanted to raise it up just in case.

Reply:
Good callout, in this case we create the `Account` from the `IdentityTokenResponse` right prior to calling this method, so the KDF in the account will be the same as in the response. That seems ok to me, but if you think we need to pass around the KDF from the response, I could change that.

Reply:
Thanks for checking it, it's perfect as it is now.